backend/dal/Services/Concrete/ParcelService.cs

```diff
@@ -28,41 +30,125 @@ public class ParcelService : BaseService<Parcel>, IParcelService
-            {
-                // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
-                query = this.Context.Parcels.AsNoTracking().Where(p =>
-                    (!p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
-                    p.Latitude <= neLat &&
-                    p.Latitude >= swLat &&
-                    p.Longitude <= neLong &&
-                    p.Longitude >= swLong);
-            }
-            if (agencyId != null)
-            {
-                query = query.Where(parcel => parcel.AgencyId == agencyId);
-            }
-            if (propertyClassificationId != null)
-            {
-                query = query.Where(parcel => parcel.ClassificationId == propertyClassificationId);
-            }
+            // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+            IQueryable<Parcel> query = this.Context.Parcels.AsNoTracking().Where(p =>
+                (!p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId))) &&
+                p.Latitude != 0 &&
+                p.Longitude != 0 &&
+                p.Latitude <= neLat &&
+                p.Latitude >= swLat &&
+                p.Longitude <= neLong &&
+                p.Longitude >= swLong);
+            return query.ToArray();
+        }
+
+        /// <summary>
+        /// Get an array of parcels within the specified filter.
+        /// Will not return sensitive parcels unless the user has the `sensitive-view` claim and belongs to the owning agency.
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public IEnumerable<Parcel> GetNoTracking(ParcelFilter filter)
+        {
+            filter.ThrowIfNull(nameof(filter));
+
+            // Check if user has the ability to view sensitive properties.
+            var userAgencies = this.User.GetAgencies();
+            var viewSensitive = this.User.HasPermission(Security.Permissions.SensitiveView);
+
+            // Users may only view sensitive properties if they have the `sensitive-view` claim and belong to the owning agency.
+            var query = this.Context.Parcels.AsNoTracking().Where(p =>
+                !p.IsSensitive || (viewSensitive && userAgencies.Contains(p.AgencyId)));
+
+            if (filter.NELatitude.HasValue && filter.NELongitude.HasValue && filter.SWLatitude.HasValue && filter.SWLongitude.HasValue)
+                query = query.Where(p =>
+                    p.Latitude != 0 &&
+                    p.Longitude != 0 &&
+                    p.Latitude <= filter.NELatitude &&
+                    p.Latitude >= filter.SWLatitude &&
+                    p.Longitude <= filter.NELongitude &&
+                    p.Longitude >= filter.SWLongitude);
+
+            if (filter.Agencies?.Any() == true)
```

Comment:
Do we need to worry about supporting child agencies here?

Reply:
I thought about that briefly.  I think we probably may.  I wasn't certain yet, so I didn't think I'd implement it in this PR.